logstash-core/src/main/java/org/logstash/common/io/PageIO.java

```diff
@@ -10,10 +10,20 @@
 public interface PageIO extends Closeable {
 
     // the concrete class should be constructed with the pageNum, capacity and dirPath attributes
+    // and open/recover/create must first be called to setup with physical data file
+    //
+    // TODO: we should probably refactor this with a factory to force the creation of a fully
```

Comment:
well, it is not really related to this PR and non-critical - the other refactors were triggered by extremely non-DRY code which I think was justified while working on this. I think we should open a followup issue